public/components/wz-kibana-vis/wz-kibana-vis.tsx

```diff
@@ -0,0 +1,187 @@
+    initializeVis(visID, tab);
+  }, []);
+
+  useEffect(() => {
+    if (currentRawVis) {
+      validateVisualization(currentRawVis);
+    }
+  }, [currentRawVis]);
+
+  /**
+   *
+   * @param visID
+   * @param tab
+   */
+  const initializeVis = async (visID: string, tab: string) => {
+    setIsLoading(true);
+    try {
+      if (!visID || !tab) {
+        throw new Error(`visID and tab are required`);
+      }
+      // assign current tab
+      tabVisualizations.removeAll();
+      tabVisualizations.setTab(tab);
+      tabVisualizations.assign({
+        general: 1,
+      });
+      // create filter handler for visualizations
+      const filterHandler = new FilterHandler(AppState.getCurrentPattern());
+      // set visFactoryHandler
+      // set rawVisualizations json definitions from api
+      if (tab === VIS_TAB_OVERVIEW) {
+        await VisFactoryHandler.buildOverviewVisualizations(filterHandler, tab, null);
+      } else if (tab === VIS_TAB_AGENTS) {
+        await VisFactoryHandler.buildAgentsVisualizations(filterHandler, tab, null);
+      } else {
+        // tab doesn't exist
+        throw new Error(`tab ${tab} doesn't exist`);
+      }
+
+      // get raw vis definition
+      const rawVis = getRawVisualizationById(visID);
+      setCurrentRawVis(rawVis);
+      if (!rawVis) {
+        throw new Error(`id ${visID} visualization is not defined`);
+      }
+    } catch (error) {
+      const options = {
+        context: `${WzKibanaVis.name}.initializeVis`,
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        store: true,
+        error: {
+          error: error,
+          message: error.message || error,
+          title: `Could not render visualization`,
+        },
+      };
+      getErrorOrchestrator().handleError(options);
+      setIsLoading(false);
+      setVisCouldBeRendered(false);
+    }
+  };
+
+  /**
+   * 
+   * @param raw 
+   */
+  const validateVisualization = async (raw) => {
+    setIsLoading(true);
+    try {
+      if (!raw) {
+        throw new Error('raw must be defined');
+      }
+
+      const visIndexPattern = getVisIndexPattern(raw);
+      if (!visIndexPattern) {
+        throw new Error('Index pattern is not defined in visualization');
+      }
+      // checking if module is enable in confi
+      await checkVisIndexExist(visIndexPattern);
+      // set state visualization is ok for render
+      setVisCouldBeRendered(true);
+      setIsLoading(false);
+    } catch (error) {
+      // visualization couldn't be rendered
+      //console.log(`Error validating visualization: ${error}`);
+      setErrorMessage(error);
+      setVisCouldBeRendered(false);
+      setIsLoading(false);
+    }
+  };
+
+  /**
+   *
+   * @param index
+   */
+  const checkVisIndexExist = async (index: string) => {
+    let res = await GenericRequest.request(
+      'GET',
+      `/api/index_patterns/_fields_for_wildcard?pattern=${index}`,
+      {}
+    );
+    return true;
```

Comment:
**nitpick**: The `res` variable is not used, we could return the property of response we need instead of a literal `true`.